src/database/GameInstance.ts

```diff
@@ -175,23 +207,28 @@ export class GameInstance {
     this.teams["RED"] = [];
   }
 
-  public shuffleTeams(shuffleMethod: "random") {
-    switch (shuffleMethod) {
-      case "random":
-        {
-          const shuffled = this.getPlayers().sort(() => Math.random() - 0.5);
-          const half = Math.ceil(shuffled.length / 2);
-
-          const blue = shuffled.slice(0, half);
-          const red = shuffled.slice(half);
-
-          this.teams.BLUE = blue;
-          this.teams.RED = red;
-        }
+  public createTeams(createMethod: "random") {
+    switch (createMethod) {
+      case "random": {
+        const simulatedTeams = this.simulateShuffledTeams();
+        this.teams.BLUE = simulatedTeams.BLUE;
+        this.teams.RED = simulatedTeams.RED;
         break;
+      }
     }
   }
 
+  public simulateShuffledTeams(): Record<Team, PlayerInstance[]> {
+    const undecidedPlayers = Array.from(this.getPlayersOfTeam("UNDECIDED"));
+    const shuffled = undecidedPlayers.sort(() => Math.random() - 0.5);
+    const half = Math.ceil(shuffled.length / 2);
+
+    return {
+      BLUE: shuffled.slice(0, half),
+      RED: shuffled.slice(half),
+    };
+  }
+
```

Comment:
i moved this logic from GameInstance into RandomTeamPickingSession in my branch, i think it's better suited there